src/core/server/saved_objects/service/lib/namespace.ts

```diff
@@ -0,0 +1,50 @@
+/*
```

Comment:
> I'm not entirely sold on this just yet. I don't love the distinction between a "namespace id" and a "namespace string".
> ...
> All that said, I like how this consolidates the previously ugly translation logic, but I question the benefit of exporting this as part of core's public contract.

My and @azasypkin's original reasoning for exporting this as part of core's public contract is that saved object operations all return an object's `namespaces` now.

An example scenario: you `find` a bunch of objects across all namespaces, and then you want to do take some follow-up CRUD action based on those results; we don't allow consumers to use `'default'` for the Default namespace, they must instead use `undefined`. So, as a consumer, I need some way to translate the results, even if the Spaces plugin is disabled.

That said, _our_ immediate use case of rotating saved objects' encryption keys doesn't require that to be exposed, because this PR also allows `bulkUpdate` across namespaces. It just made sense to us to make these two changes at the same time, since they're both intended to clean up and standardize the saved objects client API usage.

Instead of exposing these methods publicly, we could consider this: for each repository method, convert `options.namespace` to its "namespace ID" format. Then consumers shouldn't need to differentiate between `'default'` and `undefined`.

------------

As far as naming is concerned: I just needed some way to differentiate between the two if we wanted to expose these as part of core's public contract. I'm not married to "namespace ID" and "namespace string", but there is a clear difference and I couldn't think of a better name for the methods. What do you think about this instead?

`namespaceIdToString` -> `normalizeNamespace`
`namespaceStringToId` -> `namespaceIdPrefix`

Reply:
> Instead of exposing these methods publicly, we could consider this: for each repository method, convert options.namespace to its "namespace ID" format. Then consumers shouldn't need to differentiate between 'default' and undefined.

I like this idea, if it's not too much work

Reply:
> I like this idea, if it's not too much work

I removed these methods from the public contract in 0b2f62cbc8, I'll follow up with another commit to normalize namespace options as described!

Reply:
Normalized namespace options in 46bc77d41e.